apiv2/internal/server/customconfig.go

```diff
@@ -6,41 +6,149 @@ package server
 import (
 	"context"
 
-	"google.golang.org/grpc/codes"
-
-	customconfig_v1 "github.com/open-edge-platform/infra-core/apiv2/v2/internal/pbapi/resources/customconfig/v1"
+	customconfigv1 "github.com/open-edge-platform/infra-core/apiv2/v2/internal/pbapi/resources/customconfig/v1"
 	restv1 "github.com/open-edge-platform/infra-core/apiv2/v2/internal/pbapi/services/v1"
+	inv_computev1 "github.com/open-edge-platform/infra-core/inventory/v2/pkg/api/compute/v1"
+	inventory "github.com/open-edge-platform/infra-core/inventory/v2/pkg/api/inventory/v1"
 	"github.com/open-edge-platform/infra-core/inventory/v2/pkg/errors"
+	"github.com/open-edge-platform/infra-core/inventory/v2/pkg/validator"
 )
 
+func toInvCustomConfig(customConfig *customconfigv1.CustomConfigResource) (*inv_computev1.CustomConfigResource, error) {
+	if customConfig == nil {
+		return &inv_computev1.CustomConfigResource{}, nil
+	}
+
+	invCustomConfig := &inv_computev1.CustomConfigResource{
+		Name:        customConfig.GetName(),
+		Description: customConfig.GetDescription(),
+		Config:      customConfig.GetConfigContent(),
+	}
+
+	err := validator.ValidateMessage(invCustomConfig)
+	if err != nil {
+		zlog.InfraErr(err).Msg("Failed to validate inventory custom config resource")
+		return nil, err
+	}
+
+	return invCustomConfig, nil
+}
+
+func fromInvCustomConfig(invCustomConfig *inv_computev1.CustomConfigResource) *customconfigv1.CustomConfigResource {
+	if invCustomConfig == nil {
+		return &customconfigv1.CustomConfigResource{}
```

Comment:
If the received pointer is nil, we should return an error

Reply:
This is the correct behaviour. The result would be a returned `not found` from inventory if otherwise. This is an empty resource in all resources as some of them are parsed from others (e.g., parent region) and not always present.